packages/plugins/operation-headers/src/index.ts

```diff
@@ -0,0 +1,32 @@
+import type { MeshServeContext, MeshServePlugin } from '@graphql-mesh/serve-runtime';
+import type { MeshFetchRequestInit } from '@graphql-mesh/types';
+import { getHeadersObj, mapMaybePromise } from '@graphql-mesh/utils';
+
+export interface OperationHeadersFactoryPayload {
+  context: MeshServeContext;
+  url: string;
+  options: MeshFetchRequestInit;
+}
+
+export type OperationHeadersFactory = (
+  payload: OperationHeadersFactoryPayload,
+) => Promise<Record<string, string>> | Record<string, string>;
+
+export function useOperationHeaders(factoryFn: OperationHeadersFactory): MeshServePlugin {
+  return {
+    onFetch({ url, options, context }) {
+      const existingHeaders = getHeadersObj(options.headers || {});
+      const newHeaders$ = factoryFn({
+        url,
+        options,
+        context,
+      });
+      return mapMaybePromise(newHeaders$, newHeaders => {
```

Comment:
Now with #6608, you can just use `setOptions`.

Reply:
Just rebased it :thumbsup: